edb/schema/modules.py

```diff
@@ -73,3 +75,26 @@ class AlterModule(ModuleCommand, sd.AlterObject[Module]):
 
 class DeleteModule(ModuleCommand, sd.DeleteObject[Module]):
     astnode = qlast.DropModule
+
+    def _validate_legal_command(
+        self,
+        schema: s_schema.Schema,
+        context: sd.CommandContext,
+    ) -> None:
+        super()._validate_legal_command(schema, context)
+
+        # For now, we disallow deleting non-empty modules.
+
+        # Modules aren't actually stored with any direct linkage
+        # to the objects in them, so explicitly search for objects
+        # in the module (excluding the module itself).
+        has_objects = bool(any(schema.get_objects(
```

Comment:
I don't think you need `any` here

Reply:
No, you do: it returns an iterator, and `bool` on an iterator is `True`. There's no need to be clever here, though; I'll just make it `list`

Reply:
Ah, right. Let's leave any in then. Constructing a list seems wasteful, since there could be lots of objects in a module